Trying:
    lazy_matrix_mul = __import__('101-lazy_matrix_mul').lazy_matrix_mul
Expecting nothing
ok
Trying:
    print(lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]]))
Expecting:
    [[ 7, 10]
     [15, 22]]
**********************************************************************
File "./tests/101-lazy_matrix_mul.txt", line 9, in 101-lazy_matrix_mul.txt
Failed example:
    print(lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]]))
Expected:
    [[ 7, 10]
     [15, 22]]
Got:
    [[ 7 10]
     [15 22]]
Trying:
    lazy_matrix_mul('abc', [[1, 2], [3, 4]])
Expecting:
    Traceback (most recent call last):
    ValueError: Scalar operands are not allowed, use '*' instead
ok
Trying:
    lazy_matrix_mul([[1, 2], [3, 4]], 1)
Expecting:
    Traceback (most recent call last):
    ValueError: Scalar operands are not allowed, use '*' instead
ok
Trying:
    lazy_matrix_mul([[1, 2], 1], [[1, 2], [3, 4]])
Expecting:
    Traceback (most recent call last):
    ValueError: setting an array element with a sequence.
ok
Trying:
    lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], 1])
Expecting:
    Traceback (most recent call last):
    ValueError: setting an array element with a sequence.
ok
Trying:
    lazy_matrix_mul([], [[1, 2], [3, 4]])
Expecting:
    Traceback (most recent call last):
    ValueError: shapes (0,) and (2,2) not aligned: 0 (dim 0) != 2 (dim 0)
ok
Trying:
    lazy_matrix_mul([[1, 2], [3, 4]], [[]])
Expecting:
    Traceback (most recent call last):
    ValueError: shapes (2,2) and (1,0) not aligned: 2 (dim 1) != 1 (dim 0)
ok
Trying:
    lazy_matrix_mul([[1, 2], ['abc', 4]], [[1, 2], [3, 4]])
Expecting:
    Traceback (most recent call last):
    TypeError: invalid data type for einsum
ok
Trying:
    lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], ['abc', 4]])
Expecting:
    Traceback (most recent call last):
    TypeError: invalid data type for einsum
ok
Trying:
    lazy_matrix_mul([[1, 2], [1, 2, 3, 4]], [[1, 2], [3, 4]])
Expecting:
    Traceback (most recent call last):
    ValueError: setting an array element with a sequence.
ok
Trying:
    lazy_matrix_mul([[1, 2], [1, 2]], [[1, 2], [3, 4, 5, 6]])
Expecting:
    Traceback (most recent call last):
    ValueError: setting an array element with a sequence.
ok
Trying:
    lazy_matrix_mul([[1, 2, 3, 4], [1, 2, 3, 4]], [[1, 2], [3, 4]])
Expecting:
    Traceback (most recent call last):
    ValueError: shapes (2,4) and (2,2) not aligned: 4 (dim 1) != 2 (dim 0)
ok
Trying:
    lazy_matrix_mul([[1, 2], [3, 4]])
Expecting:
    Traceback (most recent call last):
    TypeError: lazy_matrix_mul() missing 1 required positional argument: 'm_b'
ok
Trying:
    lazy_matrix_mul()
Expecting:
    Traceback (most recent call last):
    TypeError: lazy_matrix_mul() missing 2 required positional arguments: 'm_a' and 'm_b'
ok
**********************************************************************
1 items had failures:
   1 of  15 in 101-lazy_matrix_mul.txt
15 tests in 1 items.
14 passed and 1 failed.
***Test Failed*** 1 failures.
